### DISCUSSION TITLE
too many positional arguments were provided in 'step ca certificate <subject> <crt-file> <key-file>'

### DISCUSSION THREAD

**Discussion by MP-ICT-Diensten** [smallstep/certificates#997] [General] [answered]
Hello,

I execute this command:
step ca certificate create example.com example.com.crt example.com.key

Error:
too many positional arguments were provided in 'step ca certificate <subject> <crt-file> <key-file>'

I finally find out that the above command does not work, it must be and that the `<crt-file>` may not be included in the command line:
step ca certificate create example.com example.com.key

This is copy/paste from the manual page:
https://smallstep.com/docs/step-ca/basic-certificate-authority-operations/#get-a-certificate

# step version
Smallstep CLI/0.21.0 (linux/amd64)
Release Date: 2022-07-06T22:23:54Z

**Comment by maraino** [comment, marked as answer]
There're two different commands:
```
# To create a certificate using the online step-ca
step ca certificate example.com example.com.crt example.com.key

# To create a certificate signed by a given issuer (offline)
step certificate create --ca iss.crt --ca-key iss.key example.com example.com.crt example.com.key
```

I believe you want to first one, without the `create` extra argument.

**Reply by MP-ICT-Diensten** [reply]
Thanks
